### ISSUE TITLE
Add the original error on UploadFileInternal(async)

### ISSUE THREAD

**Issue by Fenton613** [robinrodricks/FluentFTP#758]
Hi robinrodrick,
at FtpClient_FileUpload.cs at UploadFileInternal(Async) I found the following catch statement (see below)
If an upload throws an exception, in some cases it is tried to Resume the upload.
But sometime the ResumeUpload throws an excpetion. But in this case the original error is lost.

Is it possible to add a try catch inside the catches and throw an AggregateException? So the original epception can be locked.
(Maybe there is a possibility to fix the original exception)
```
catch (IOException ex) {

						// resume if server disconnected midway, or throw if there is an exception doing that as well
						if (!ResumeUpload(remotePath, ref upStream, remotePosition, ex)) {
							sw.Stop();
							throw;
						}

						// since the remote stream has been seeked, we need to reposition the local stream too
						if (fileData.CanSeek) {
							fileData.Seek(localPosition, SeekOrigin.Begin);
						}
						else {
							sw.Stop();
							throw;
						}

```

By the way, thank  you very much for FluentFTP and the nice updates.

**Comment by robinrodricks** [comment]
I will look into it.

**Comment by robinrodricks** [comment]
Hi, can you tell me where exactly the exception is thrown? I cannot find any such thing. You said ResumeUpload throws an exception, in that case, which error is lost? The `ex` object or the exception thrown by ResumeUpload?

**Comment by robinrodricks** [comment]
Is this what you want?

https://github.com/robinrodricks/FluentFTP/commit/ba9077bd1289f27ca2dcc28acd3b310cbfe58560

**Comment by robinrodricks** [comment]
Released in version 38.0.0.

https://www.nuget.org/packages/FluentFTP/38.0.0

**Comment by robinrodricks** [comment]
Closing this, let me know in a new ticket if there are issues.
